libraries/AP_RangeFinder/AP_RangeFinder_MAVLink.cpp

```diff
@@ -27,13 +27,14 @@ void AP_RangeFinder_MAVLink::handle_msg(const mavlink_message_t &msg)
     mavlink_distance_sensor_t packet;
     mavlink_msg_distance_sensor_decode(&msg, &packet);
 
-    // only accept distances for the configured orentation
+    // only accept distances for the configured orientation
     if (packet.orientation == orientation()) {
         state.last_reading_ms = AP_HAL::millis();
         distance_cm = packet.current_distance;
         _max_distance_cm = packet.max_distance;
         _min_distance_cm = packet.min_distance;
         sensor_type = (MAV_DISTANCE_SENSOR)packet.type;
+        signal_quality = packet.signal_quality;
```

Comment:
This is not interpretting the mavlink field correctly:
```
      <field type="uint8_t" name="signal_quality" units="%">Signal quality of the sensor. Specific to each sensor type, representing the relation of the signal strength with the target reflectivity, distance, size or aspect, but normalised as a percentage. 0 = unknown/unset signal quality, 1 = invalid signal, 100 = perfect signal.</field>
```

Reply:
so, signal quality 0 means mavlink has no data, so it maps to -1 in ardupilot, while 1 is the worst signal and is mapped to 0, does that make sense?